plugins/org.jkiss.dbeaver.model.ai/src/org/jkiss/dbeaver/model/ai/completion/DAICompletionEngine.java

```diff
@@ -19,46 +19,52 @@
 
 import org.jkiss.code.NotNull;
 import org.jkiss.dbeaver.DBException;
-import org.jkiss.dbeaver.model.ai.format.IAIFormatter;
+import org.jkiss.dbeaver.model.ai.AISettingsEventListener;
+import org.jkiss.dbeaver.model.ai.n.AIStreamingResponseHandler;
+import org.jkiss.dbeaver.model.data.DBDObject;
 import org.jkiss.dbeaver.model.runtime.DBRProgressMonitor;
 
 import java.util.List;
-import java.util.Map;
 
 /**
  * Completion engine
  */
-public interface DAICompletionEngine<SERVICE> {
+public interface DAICompletionEngine extends AISettingsEventListener {
 
     /**
      * Completion engine name
      */
+    @NotNull
     String getEngineName();
 
-    /**
-     * Do query completion
-     */
-    @NotNull
-    List<DAICompletionResponse> performQueryCompletion(
+    int getContextWindowSize(@NotNull DBRProgressMonitor monitor);
+
+    void chat(
+        @NotNull DBRProgressMonitor monitor,
+        @NotNull DAICompletionContext context,
+        @NotNull List<DAIChatMessage> messages,
+        @NotNull AIStreamingResponseHandler handler
+    );
+
+    void describe(
+        @NotNull DBRProgressMonitor monitor,
+        @NotNull DAICompletionContext context,
+        @NotNull DBDObject toDescribe,
```

Comment:
There may be multiple objects to describe.
There must be some options (like brief/detailed description) and maybe some user comment too.